forest/bonsai/simulate_gps_data.py

```diff
@@ -1035,6 +1043,111 @@ def prepare_data(
     )
 
 
+def process_attributes(
```

Comment:
This function is doing error checking/data munging for `Attributes` objects. I think this would be best accomplished by converting `Attributes` to a regular class and moving this logic to the constructor.

The logic for creating the `switches` data structure should be kept separately. Could you bring in some code that shows the use of `switches` to help understand the design?

Reply:
So the `switches` are used to change the `active_status` or the `travelling_status` in the middle of generating their trajectories. It is used in `simulate_gps` module in lines `928-936` and `954-957`.

Also to check the format of the attributes dictionary which will be loaded from a json file in the final function check `sample_attributes` in lines `430-459` in the `test_simulate_gps` module.